common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -5561,7 +5563,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "Size of the runtime statistics cache. Unit is: OperatorStat entry; a query plan consist ~100."),
     HIVE_QUERY_PLANMAPPER_LINK_RELNODES("hive.query.planmapper.link.relnodes", true,
         "Whether to link Calcite nodes to runtime statistics."),
-
+    HIVE_QUERY_MAX_RECOMPILATION_COUNT("hive.query.recompilation.max.count", 1,
+        "Maximum number of re-compilations for a single query."),
```

Comment:
Why do we need the number of recompilations to be configurable?

Reply:
That was only my feeling for symmetry. I could remove, and always do it only once. I am not sure the same does not apply to the `HIVE_QUERY_MAX_REEXECUTION_COUNT`.

Reply:
I tend to avoid adding new toggles unless it is necessary. They require testing and become part of the API that we need to maintain and support.